btcec/schnorr/musig2/context.go

```diff
@@ -554,15 +554,31 @@ func (s *Session) RegisterPubNonce(nonce [PubNonceSize]byte) (bool, error) {
 func (s *Session) Sign(msg [32]byte,
 	signOpts ...SignOption) (*PartialSignature, error) {
 
-	switch {
 	// If no local nonce is present, then this means we already signed, so
 	// we'll return an error to prevent nonce re-use.
-	case s.localNonces == nil:
+	if s.localNonces == nil {
 		return nil, ErrSigningContextReuse
+	}
+
+	opts := defaultSignOptions()
```

Comment:
One alternative would be to add a method to the session to add the combined nonce. 

Then we can leave this error path as is (failed if no combined nonce is present).